### PR TITLE
Beta v3

### LINKED ISSUE #5: Update the README and make an About page
If someone could mess around with the app and do a write up of the use cases in the README so I don't have to that would be awesome. For the about page, I want to put in a blurb about why I made the app and put pictures and infor of the contributors, if you want to contribute to the app feel free to add your info in there.

### LINKED ISSUE #13: Professor and school administrator takedown button
Have a button somewhere on the app (probably user settings) that allow the professor or a dean / department chair to request removal of a syllabus. When this is done, hide the syllabus from public view, but do not delete it until an admin can verify the authority of who asked for it to be taken down.

This is hipri

### LINKED ISSUE #18: Better background image
find a better background image. Would prefer a lighter image

### LINKED ISSUE #19: Parse Professors Names better
Some professors do not have names that follow 'FirstName Lastname' structure. For example Prof. Seung-Hun Lee has a hyphen which is getting caught and blocked by the professor name filter. 

Need to edit line 175 of views.py to be less strict

### LINKED ISSUE #31: Update README with tech used for better visibility
Also update the app description to include "Django" app.

It might be better to include screenshots and also a disclaimer that only .edu email works for now (any only Gsuite users?).

### PR DESCRIPTION
**Pull request by verndrade**
- Added AdminView, which allows admins to see the views of different schools without creating new superusers
- Made about page and added team members
- Made a school takedown notice page, with reasoning attached. Manual setting is necessary for now
- Updated README to help users understand how to use the app
- Updated background image
- Improved professor name validation
- admin feedback removal

### CODE REVIEW

File: syllabiShare/views.py [hunk 1]
```diff
@@ -173,15 +198,15 @@ def upload(request):
     success = False
     message = 'Misuse of uploads will be met by a ban!'
     if request.method == 'POST':
-        course = request.POST['prof'].split()
-        goodProf = len(request.POST['prof'].split()) == 2 and course[0].isalpha() and course[1].isalpha()
+        prof = request.POST['prof'].strip().split()
+        goodProf = len(prof) == 2 and all(char.isalpha() or char == '-' or char == '\'' for char in prof[0]) and all(char.isalpha() or char == '-' or char == '\'' for char in prof[1])
```
Comment (winstliu) [inline syllabiShare/views.py:205]: Will fail on e.g. `S Travis Elliott` or non-Latin names. Consider just letting them put whatever they want.

File: templates/error.html [hunk 2]
```diff
@@ -12,15 +12,15 @@
...
     <b style="top: 30%; font-size: 50px;"> WELCOME TO SYLLABI SHARE! </b> <br><br>
     <b style="top: 60%; font-size: 30px;"> Learn about your courses here! </b> <br>
     <button class="btn-dark btn" style="font-size: 30px; display: block; margin: auto;"> <b> <a style="color:antiquewhite;" href="{% url 'social:begin' 'google-oauth2' %}"> Please Login with a .edu Email! </a> </b> </button> <br>
     
     <div class="text-center">
-      <h6>By logging in to this website you are agreeing to the <a href="/privacy" style="color: blueviolet"> privacy policy</a></h6> <br>
+      <h6>By logging in to this website you are agreeing to the <a href="/privacy" style="color: purple"> privacy policy </a> and certify that you will only use the material for educational purposes</h6> <br>
       {% if loggedIn %}
-      <div class="alert alert-danger" style="width: 20%; margin: auto;" role="alert"> That was not a .edu email!</div>
+      <div class="alert btn-danger" style="width: 20%; margin: auto;" role="alert"> That was not a .edu email!</div>
```
Comment (winstliu) [inline templates/error.html]: Wait, shouldn't this still be alert?

File: templates/upload.html [hunk 3]
```diff
@@ -26,18 +26,20 @@ <h4 style='float: right;'> PDFs only</b> </h4>
 </form>
 
 {% if success %}
-<div id="message" class="alert alert-success text-center" style="width: 40%; margin: auto; margin-top: 5px;" role="alert"> Addition Successful! </div>
+<div id="message" class="alert btn-success text-center" style="width: 40%; margin: auto; margin-top: 5px;" role="alert"> Addition Successful! </div>
```
Comment (winstliu) [inline templates/upload.html]: I can't find the commit that changed these at a glance, but it feels to me like these alert->btn changes were a mistake.

**Review by winstliu — COMMENTED** [review]
Curious about the alert->btn conversions.
Rest is fine.

File: templates/upload.html (hunk 3 above)
Comment (winstliu) [reply templates/upload.html]: Yeah, these definitely need to be alert. `btn` makes them appear clickable.

File: syllabiShare/views.py (hunk 1 above)
Comment (verndrade) [reply syllabiShare/views.py:205]: is that a professors name? cant find it anywhere.

Comment (winstliu) [reply syllabiShare/views.py:205]: His full legal name is Sean Travis Elliott, but he goes by his middle name. Prefers to be referred to as S. Travis Elliott in official documents.

So maybe not the *best* example, but then there's also `Mohammad Mahmoody Ghidary` in the UVA CS department and a bunch of more examples in the Spanish department.

(Also, I believe the `isalpha` check will fail for [`Fernando Operé`](http://spanitalport.as.virginia.edu/people/fo))

Comment (verndrade) [reply syllabiShare/views.py:205]: I can see that but its pretty trivial if students can still figure out who the profs are. if we get a report we can change it pretty easily on the back end. Leaving it free for input can lead to some shenanigans. isalpha accounts for accented letters
image

File: templates/upload.html (hunk 3 above)
Comment (verndrade) [reply templates/upload.html]: fixed

File: syllabiShare/views.py (hunk 1 above)
Comment (winstliu) [reply syllabiShare/views.py:205]: Fair.

File: templates/upload.html [hunk 4]
```diff
@@ -21,23 +21,39 @@ <h4 style='float: left;' for="file"> <b>Syllabus:</b> </h4>
       <h4 style='float: right;'> PDFs only</b> </h4>
       <input style="width:100%;" id="file" type="file" name="file" accept="application/pdf" required> <br>
     </div>
+    <div class="form-group group">
+      <h4 style='float: left;' for="file"> <b>Semester:</b> </h4>
+      <select name="semester" id="semester" style="margin: 1%;" required>
```
Comment (winstliu) [inline templates/upload.html]: january?

File: templates/upload.html [hunk 5]
```diff
@@ -21,23 +21,39 @@ <h4 style='float: left;' for="file"> <b>Syllabus:</b> </h4>
...
+        <option value="fall">Fall</option>
+        <option value="spring">Spring</option>
+        <option value="summer">Summer</option>
+      </select>
+      <select name="year" id="year" required>
```
Comment (winstliu) [inline templates/upload.html]: You're going to want to either make this a free-form input (that only accepts numbers), or have some way to automatically update the values so you don't have to come in here every year to add another option.

**Review by winstliu — APPROVED** [review]
Fine by me

File: templates/upload.html (hunk 4 above)
Comment (verndrade) [reply templates/upload.html]: ah Jterms my fatal weakness. will add before merge

File: syllabiShare/views.py [hunk 6]
```diff
@@ -173,19 +198,23 @@ def upload(request):
...
-            entry.prof = request.POST['prof']
+            entry.prof = prof[0] + ' ' + prof[1]
             entry.course = request.POST['course'].upper()
+            entry.title = request.POST['title']
             entry.dept = course[0].upper()
+            entry.semester = request.POST['semester']
+            entry.year = request.POST['year']
             entry.upvotes = 1
             entry.syllabus = request.FILES['file']
+            entry.syllabus.name = '_'.join([prof[0].lower(), prof[1].lower(), course[0], course[1], entry.semester, entry.year])
```
Comment (winstliu) [inline syllabiShare/views.py]: Add a `.pdf` to this.

**Review by winstliu — CHANGES_REQUESTED** [review]
Noticed that all the files were missing a PDF extension when testing changes out locally.

File: syllabiShare/views.py (hunk 6 above)
Comment (verndrade) [reply syllabiShare/views.py]: i thought the pdf is automatically added lmao my bad

Comment (verndrade) [reply syllabiShare/views.py:220]: oop i lied ill do that

**Review by winstliu — APPROVED** [review]

**Review by winstliu — APPROVED** [review]
Again

**Pull request merged**